test/TestGrainInterfaces/VersionAwarePlacementDirector.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.Placement;
+using Orleans.Runtime;
+using Orleans.Runtime.Placement;
+
+namespace UnitTests.GrainInterfaces
+{
+    [AttributeUsage(AttributeTargets.Class)]
+    public sealed class VersionAwareStrategyAttribute : PlacementAttribute
+    {
+        public VersionAwareStrategyAttribute()
+            : base(VersionAwarePlacementStrategy.Singleton)
+        {
+        }
+    }
+
+    [Serializable]
+    public class VersionAwarePlacementStrategy : PlacementStrategy
+    {
+        internal static VersionAwarePlacementStrategy Singleton { get; } = new VersionAwarePlacementStrategy();
+
+        private VersionAwarePlacementStrategy()
+        { }
+
+        public override bool Equals(object obj)
+        {
+            return obj is VersionAwarePlacementStrategy;
+        }
+
+        public override int GetHashCode()
+        {
+            return GetType().GetHashCode();
+        }
+    }
+
+    public class VersionAwarePlacementDirector : IPlacementDirector<VersionAwarePlacementStrategy>
+    {
+        private readonly Random random = new Random();
```

Comment:
SafeRamdom is not public. And since it is a director only used in a test, that does not create activation in parallel, I did not reimplemented it